### PR TITLE
Added fireEvent(Event<?>) overload to address issue #668

### PR DESCRIPTION
**Pull request by timeu**

### CODE REVIEW

File: gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java [hunk 1]
```diff
@@ -225,6 +226,10 @@ public void fireEvent(GwtEvent<?> event) {
         getEventBus().fireEventFromSource(event, this);
```
Comment (rdwallis) [inline gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java]: `public void fireEvent(GwtEvent<?> event)`

seems a bit redundant now.

Is there a web bindery replacement for HasHandlers?

Comment (timeu) [inline gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java]: Unfortunately there seems to be no replacement: 

http://www.gwtproject.org/javadoc/latest/com/google/web/bindery/event/shared/package-summary.html

File: gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java [hunk 2]
```diff
@@ -225,6 +226,10 @@ public void fireEvent(GwtEvent<?> event) {
...
     }
 
+    public void fireEvent(Event<?> event) {
```
Comment (rdwallis) [inline gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java]: Please add javadoc.  (Just copy from HasHandlers)

File: gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java (hunk 1 above)
Comment (rdwallis) [inline gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java]: change:
`getEventBus().fireEventFromSource(event, this);`
to:
`fireEvent((Event<?>) event);`

Then any future changes to fire event can be made in the one method.

Comment (timeu) [inline gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java]: OK I did the changes as requested (I did commit --amend to make sure that we don't pollute the commit history). Hope this is fine

**Comment by rdwallis** [comment]
Thanks for the contribution!

LGTM @Chris-V @christiangoudreau ?

**Comment by christiangoudreau** [comment]
Looks good to me!

**Comment by Chris-V** [comment]
LGTM. Don't forget the javadoc, then I merge :)

**Pull request merged**